src/wallet/wallet.ts

```diff
@@ -275,24 +305,44 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
     this.clearSensitiveData();
   }
 
-  generateCreateWalletAuthData(seed: string): CreateWalletAuthData {
-    const xpriv = walletUtils.getXPrivKeyFromSeed(seed, {passphrase: this.passphrase, networkName: this.network.name});
-    const xpub = walletUtils.getXPubKeyFromSeed(seed, {passphrase: this.passphrase, networkName: this.network.name});
-    const authXpub = HathorWalletServiceWallet.getAuthXPubKeyFromSeed(seed, {
-      passphrase: this.passphrase,
-      networkName: this.network.name,
-    });
+  generateCreateWalletAuthData(pinCode: string): CreateWalletAuthData {
+    let xpriv: bitcore.HDPrivateKey;
+    let xpub: string;
+    let authXpub: string;
+    let xprivAccountPath: string;
+    let authDerivedPrivKey: bitcore.HDPrivateKey;
 
     const now = new Date();
     const timestampNow = Math.floor(now.getTime() / 1000); // in seconds
-    const walletId = HathorWalletServiceWallet.getWalletIdFromXPub(xpub);
+
+    if (this.seed) {
+      xpriv = walletUtils.getXPrivKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
+      xpub = walletUtils.getXPubKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
```

Comment:
I would just add a comment here and above.

`getXPubKeyFromSeed` gets the xpub in the account derivation path while `getXPrivKeyFromSeed` gets the privateKey in the root derivation path.